pkg/build/builder/docker.go

```diff
@@ -83,21 +84,35 @@ func (d *DockerBuilder) Build() error {
 		push = true
 	}
 
-	if err := d.dockerBuild(buildDir, d.build.Spec.Source.Secrets); err != nil {
+	buildTag := randomBuildTag(d.build.Namespace, d.build.Name)
+
+	if err := d.dockerBuild(buildDir, buildTag, d.build.Spec.Source.Secrets); err != nil {
+		return err
+	}
+
+	cname := containerName("docker", d.build.Name, d.build.Namespace, "post-commit")
+	if err := execPostCommitHook(d.dockerClient, d.build.Spec.PostCommit, buildTag, cname); err != nil {
```

Comment:
We could remove it, but just like we decided not to remove the `pushTag` anymore, I indeed think it's worth leaving it around. If we keep it, running a consecutive build can reuse layers from cache.

We only untag (`removeImage(d.dockerClient, buildTag)`) if we've successfully tagged the image with `pushTag`.

That means that consecutive builds can always reuse layers from the cache, no matter if the post commit hook succeeded or not.

Should we do it differently?